frigate/http.py

```diff
@@ -446,10 +449,37 @@ def latest_frame(camera_name):
         return "Camera named {} not found".format(camera_name), 404
 
 
+@bp.route("/vod/<path:path>")
+def vod(path):
+    if not os.path.isdir(f"{RECORD_DIR}/{path}"):
+        return "Recordings not found.", 404
+
+    files = glob.glob(f"{RECORD_DIR}/{path}/*.mp4")
+    files.sort()
+
+    clips = []
+    durations = []
+    for filename in files:
+        clips.append({"type": "source", "path": filename})
+        video = cv2.VideoCapture(filename)
+        duration = int(
+            video.get(cv2.CAP_PROP_FRAME_COUNT) / video.get(cv2.CAP_PROP_FPS) * 1000
```

Comment:
How long does it take to do this dynamically? It could take a while if using a network drive.

Reply:
It’s two seconds on my dev machine, but it’s also pretty beefy and the files were local. I almost just hardcoded the 60 seconds here, but know that the command (and thus the time) can be changed in configuration.

I saw the use of ffprobe elsewhere for clips, but since OpenCV was already imported and [this stackoverflow answer](https://stackoverflow.com/a/61572332) said it was faster, I just used it. Granted I did not do any verification myself. It’s probably worth a shot testing this on a Pi with video on a network drive to see.

Reply:
Verified timing in comment below.